plugins/destination/mssql/client/migrate.go

```diff
@@ -2,48 +2,190 @@ package client
 
 import (
 	"context"
+	"database/sql"
 	"fmt"
+	"strings"
 
+	"github.com/cloudquery/cloudquery/plugins/destination/mssql/queries"
 	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
 )
 
-// Migrate relies on the CLI/client to lock before running migration.
-func (c *Client) Migrate(ctx context.Context, tables schema.Tables) error {
-	for _, table := range tables {
-		if err := c.migrate(ctx, table); err != nil {
+func (c *Client) schemaTables(ctx context.Context, tables schema.Tables) (schema.Tables, error) {
+	query, params := queries.AllTables(c.schemaName)
+	rows, err := c.db.QueryContext(ctx, query, params...)
+	if err != nil {
+		c.logErr(err)
+		return nil, err
+	}
+
+	schemaTables := make(schema.Tables, 0)
+	if err := processRows(rows, func(row *sql.Rows) error {
+		var tableCatalog string
+		var tableName string
+		var tableType string
+		var schemaType string
+
+		if err := row.Scan(&tableCatalog, &tableType, &tableName, &schemaType); err != nil {
 			return err
 		}
+		if tables.Get(tableName) == nil {
+			return nil
+		}
+		schemaTables = append(schemaTables, &schema.Table{Name: tableName})
+		return nil
+	}); err != nil {
+		c.logErr(err)
+		return nil, err
 	}
 
-	return nil
+	for _, table := range schemaTables {
+		columns, err := c.getTableColumns(ctx, table)
+		if err != nil {
+			return nil, err
+		}
+		table.Columns = columns
+		pks, err := c.getTablePK(ctx, table)
+		if err != nil {
+			return nil, err
+		}
+		for _, pk := range pks {
+			if col := table.Columns.Get(pk); col != nil {
+				col.CreationOptions.PrimaryKey = true
+			}
+		}
+	}
+
+	return schemaTables, nil
 }
 
-func (c *Client) migrate(ctx context.Context, table *schema.Table) (err error) {
-	c.logger.Info().Str("table", table.Name).Msg("Migrating table started")
-	defer c.logger.Info().Str("table", table.Name).Msg("Migrating table done")
-	defer func() {
-		if err != nil {
-			c.logErr(err)
+func (c *Client) normalizedTables(ctx context.Context, tables schema.Tables) schema.Tables {
+	allTables := tables.FlattenTables()
+	var normalized schema.Tables
+	for _, table := range allTables {
+		tableCopy := table.Copy(table.Parent)
+		for i := range tableCopy.Columns {
+			// Since multiple schema types can map to the same MSSQL type we need to normalize them to avoid false positives when detecting schema changes
+			tableCopy.Columns[i].Type = queries.SchemaType(queries.SQLType(table.Columns[i].Type))
 		}
-	}()
+		// If there are no PKs, we use CqID as PK
+		pks := tableCopy.PrimaryKeys()
```

Comment:
This logic was moved from `GetPKColumns`. We now do it at the beginning of the migration so no need to do it again